### DISCUSSION TITLE
I have a question about shallow() behavior

### DISCUSSION THREAD

**Discussion by hi-paul** [pmndrs/zustand#2463] [Q&A] [answered]
I think this is a function that does shallow comparison, but when I print it out as log comparing object literals to tests, I always get 
true. Is this intentional behavior? If so, can you tell me why?

```typescript
// the result is true, but i expected false
console.log(shallow({}, {}));
```

**Comment by dai-shi** [comment, marked as answer]
I guess you misunderstand shallow comparison with reference comparison.

Here's an old doc about shallow comparison https://legacy.reactjs.org/docs/shallow-compare.html, but you should be able to find a better resource on the web.
